exporters/failure/collector_github.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import logging
+from datetime import datetime
+
+import pytz
+import requests
+
+import pelorus
+from failure.collector_base import AbstractFailureCollector, TrackerIssue
+
+# One query limit, exporter will query multiple times.
+# Do not exceed 100 results
+# TODO
+GITHUB_SEARCH_RESULTS = 100
+TRACKER_PROVIDER = "github"
+
+
+class GithubAuthenticationError(Exception):
+    """
+    Exception raised for authentication issues
+    """
+
+    auth_message = "Check the TOKEN: not authorized, invalid credentials"
+
+    def __init__(self, message=auth_message):
+        super().__init__(message)
+
+
+class GithubFailureCollector(AbstractFailureCollector):
+    """
+    Github implementation of a FailureCollector
+    """
+
+    REQUIRED_CONFIG = ["TOKEN"]
+    _defaultapi = "api.github.com"
+
+    def __init__(self, apikey, projects, user=None, server=None):
+        super().__init__(server, user, apikey)
+        if projects:
+            self.projects = projects.split(",")
+        else:
+            logging.warning("No projects defined for github issues")
+            self.projects = None
+        try:
+            self.user = self._get_github_user()
+        except Exception:
+            logging.warning("github username not found")
+            self.user = "unknown"
+        self.server = self._defaultapi
+        self.tracker_provider = TRACKER_PROVIDER
+
+    def _get_github_user(self):
+        # login and get username
+        # set the username / server to env for exporter consistency
+        url = "https://{}/user".format(self._defaultapi)
+        headers = {"Authorization": "token {}".format(self.apikey)}
+        resp = self._make_request(headers, url)
+        return resp["login"]
+
+    def _make_request(self, headers, url):
+        this_response = {}
+        s = requests.Session()
+        s.headers.update(headers)
+        # disable .netrc
+        s.trust_env = False
+        resp = s.get(url, headers=headers)
```

Comment:
You don't need to do both `Session.update` and `get(headers=headers)`.

It looks like the same auth token is used each time, so we can just set that in the Session setup.

The `Accept` part could be set per-request, so `headers=headers` should be used then.